vowpalwabbit/slates.cc

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include "slates.h"
+#include "example.h"
+#include "slates_label.h"
+#include "reductions.h"
+#include "global_data.h"
+
+#include "vw.h"
+#include <algorithm>
+#include "cb_algs.h"
+#include "cb_adf.h"
+#include "conditional_contextual_bandit.h"
+#include "decision_scores.h"
+#include "action_score.h"
+#include "ccb_label.h"
+
+using namespace VW::config;
+
+namespace slates
+{
+template <bool is_learn>
+void slates_data::learn_or_predict(LEARNER::multi_learner& base, multi_ex& examples)
+{
+  _stashed_labels.reserve(examples.size());
```

Comment:
is `_stashed_labels` assumed to be empty here? Weird things can happen if its not. Maybe call `clear()` just to enforce the assumption?

Reply:
`_stashed_labels` is a vector and not a v_array. But I can still add a clear